tests/components/Drawer.test.js

```diff
@@ -28,4 +48,29 @@ describe('Drawer Component', () => {
 
         expect(asFragment()).toMatchSnapshot();
     });
+
+    it('should disable/enable body scroll when drawer is toggled', () => {
+        const { getByRole } = renderWithProps();
+        const triggerButton = getByRole('button');
+
+        expect(enableBodyScroll).toHaveBeenCalled();
+
+        fireEvent.click(triggerButton);
+
+        expect(disableBodyScroll).toHaveBeenCalled();
+
+        fireEvent.click(triggerButton);
+
+        expect(enableBodyScroll).toHaveBeenCalled();
+    });
+
+    it('should not disable body scroll when lockBodyScroll prop is false', () => {
+        const { getByRole } = renderWithProps({ lockBodyScroll: false });
+        const triggerButton = getByRole('button');
+
+        fireEvent.click(triggerButton);
+
+        expect(disableBodyScroll).toHaveBeenCalledTimes(0);
+        expect(enableBodyScroll).toHaveBeenCalledTimes(0);
```

Comment:
```suggestion
        expect(disableBodyScroll).not.toHaveBeenCalled();
```

Reply:
The way I had it (`expect(enableBodyScroll).toHaveBeenCalledTimes(0)`) I was specifically making sure that these two functions were never called.

The way you are suggesting, I need to change the test logic a little bit.

#### Why?

The order is:
1. Rendering;
2. Firing click event to open the drawer;
3. Asserting that both functions were never called.

By changing the assertions to what you suggested, we are asserting that the `click event` did not fire any of the functions. But in fact, we want to test that `enableBodyScroll` was not called when the component mounts, and then we want to test that `disableBodyScroll` was not called when the click happens.

I think we have 2 options here:
1. leave it as it currently is;
2. change the order of the asserts on the test - ✅  - this option is the one I've decided to use for the sake of code readability.

Thus, the current test order is:
1. Rendering;
2. Asserting `enableBodyScroll` function was not called;
3. Firing click event to open the drawer;
4. Asserting `disableBodyScroll` function was not called.

Reply:
I like the new approach better because it's more readable indeed. Furthermore, by rearranging the order of the assertions, it becomes clearer why we're asserting what we're asserting at the point in the test's execution flow.